src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8140,130 +8157,197 @@ void checkWellKnown(WellKnownMember member)
             // 2. For Index: Has an accessible indexer with a single int parameter
             //    For Range: Has an accessible Slice method that takes two int parameters
 
-            var lookupResult = LookupResult.GetInstance();
-
-            if (TryLookupLengthOrCount(receiverType, lookupResult, out lengthOrCountProperty, ref useSiteInfo) &&
-                TryFindIndexOrRangeIndexerPattern(lookupResult, receiverOpt, receiverType, argIsIndex, out patternSymbol, diagnostics, ref useSiteInfo))
+            if (TryBindLengthOrCount(syntax, receiverPlaceholder, out lengthOrCountAccess, diagnostics) &&
+                tryBindUnderlyingIndexerOrSliceAccess(syntax, receiver, argIsIndex, out indexerOrSliceAccess, out argumentPlaceholders, diagnostics))
             {
-                CheckImplicitThisCopyInReadOnlyMember(receiverOpt, lengthOrCountProperty.GetMethod, diagnostics);
```

Comment:
>CheckImplicitThisCopyInReadOnlyMember(receiverOpt, lengthOrCountProperty.GetMethod, diagnostics); [](http://example.com/codeflow?start=16&length=97)

It looks like we should keep this line and move it to ```TryBindLengthOrCount``` instead, next to the ```CheckValue``` call. Even though it is not perfect, at least we will preserve the same level of validation as before. Once ```CheckValue``` is improved, the line can be removed.

Reply:
On the second thought. Given that there is another call site for ```TryBindLengthOrCount``` and we don't have a real receiver there. It is probably preferred to leave this line here instead of moving it into ```TryBindLengthOrCount```